vpr/src/draw/search_bar.cpp

```diff
@@ -123,8 +144,15 @@ void search_and_highlight(GtkWidget* /*widget*/, ezgl::application* app) {
     else if (search_type == "Net Name") {
         std::string net_name = "";
         ss >> net_name;
+        AtomNetId atom_net_id = AtomNetId::INVALID();
+        atom_net_id = atom_ctx.nlist.find_net(net_name);
 
-        highlight_nets((std::string)net_name);
+        if (atom_net_id == AtomNetId::INVALID()) {
+            warning_dialog_box("Invalid Net Name");
+            return; //name not exist
+        }
+        ClusterNetId clb_net_id = atom_ctx.lookup.clb_net(atom_net_id);
+        highlight_nets(clb_net_id);
```

Comment:
This is a long routine -- suggest making some helper functions for the block names, net names etc. so it's shorter.